lib/src/pick_int.dart

```diff
@@ -10,38 +10,39 @@ extension NullableIntPick on Pick {
   /// Parses the picked value as [int]. Also tries to parse [String] as [int]
   /// via [int.tryParse]
   /// {@endtemplate}
-  int _parse() {
+  int _parse(bool roundDoubleToInt) {
     final value = required().value;
     if (value is int) {
       return value;
     }
+    if (value is num && roundDoubleToInt) {
+      return value.round();
+    }
     if (value is String) {
       final parsed = int.tryParse(value);
       if (parsed != null) {
         return parsed;
       }
     }
-    throw PickException(
-        'Type ${value.runtimeType} of $debugParsingExit can not be parsed as int');
+    throw PickException('Type ${value.runtimeType} of $debugParsingExit can not be parsed as int');
```

Comment:
We should improve the error message and add a hint to use the new round/truncate parameters if neither of them is true

Reply:
done